xpk.py

```diff
@@ -5829,6 +5994,12 @@ def workload_create(args) -> None:
     0 if successful and 1 otherwise.
   """
   add_zone_and_project(args)
+
+  if args.use_pathways:
+    update_cluster_command_code = update_pathways_cluster_if_necessary(args)
+    if update_cluster_command_code != 0:
+      xpk_exit(update_cluster_command_code)
+
```

Comment:
I don't think we should run cluster update commands in `workload_create`. This mixes the intent of the two command apis.

Reply:
Thanks, I agree with the rationale. 
However, I am hoping to have a one-time update to the cluster (if needed) on the first "workload create" on an existing cluster. New clusters won't need this change.

Handling this one-time at workload create would mean that users don't have to run "cluster create" command explicitly to upgrade their cluster. (Hoping for less intervention from users.)

Happy to chat offline, in case of concerns.

Reply:
Thought I like how it reduces the steps on the user, I think this can provide confusion between what the commands do. I would like to avoid opening up workload create API to cluster changes including and beyond this change. It might work well for this feature but it opens up a lot of potential features that I don't think we should have. What do you think Roshani?

Reply:
Thanks @Obliviour, for your thoughts!
I can see a way forward where I exit from XPK workload create-pathways to ask the user to update the cluster using XPK cluster create-pathways. I can conditionally update the Pathways server and proxy addresses based on whether clouddns is enabled. However, this way needs an extra step from the user.

I would prefer to keep the current implementation, even for short term (until existing clusters are updated) and deprecate this path. This additional condition won't be needed in case of new clusters anyway.

WDYT, @shaurya89

Reply:
I went ahead with the approach I mentioned above. That would remove cluster update from workload create flow and prompt to user to update (only when needed - Pathways in headless mode).